### ISSUE TITLE
`preview-project-description` flashes when running a project after entering and leaving the editor

### ISSUE THREAD

**Issue by mybearworld** [ScratchAddons/ScratchAddons#6884]
### Description

When using `preview-project-description` on a project with speech bubbles, it'll flash whenever they appear.

### Steps to reproduce

1. Go to a project you made with speech bubbles
2. Enable the "Preview" toggle
3. Run the project

### Environment

Browser: Edge 119
Scratch Addons version: 1.35.1
Operating system: Windows

**Comment by mybearworld** [comment]
> Is this any project or is it related to the say/think/ask blocks?

Seems to those blocks only - I encountered this on a project that used them extensively, so I didn't notice

**Comment by DNin01** [comment]
Seems somewhat performance-related. I managed to replicate your issue by having a couple dozen clones repeatedly saying hello while running it at a high framerate.

https://github.com/ScratchAddons/ScratchAddons/assets/106490990/26af6c2e-259d-49ad-a720-398ad7c7482c

I was also able to get that to happen with Turbo Mode, but I needed a lot more clones.

But this is actually a result of another issue. Try selecting text on the preview while a speech bubble is visible. The text gets deselected! So this is actually happening _any time_ a speech bubble is visible, you just need to have a lot happening to see it because this bug happens so fast.

After looking into it a little more, as it turns out, this is not specific to the speech bubbles, and I can't figure out exactly why it's happening.
- In the project I showcased, it happened while speech bubbles were on the screen, but only after I entered and left the editor.
- In various other projects, this started happening several times for no apparent reason, again, only after I entered and left the editor. (That seems to be a common theme.)
- In one project, it happened whenever a variable monitor updated.
- It also seems to happen whenever the project starts or stops running, and several times in quick succession when you first start the project.

Oh! The console, yes! Every time this happens, a `[SA] preview-project-description [page]` › `Returning a fake username to mapStateByProps` message appears, logged by line 214 of this addon's userscript.js file. Here's the surrounding code:

https://github.com/ScratchAddons/ScratchAddons/blob/3a342fa4a27cc79f138df666a03d72b4a548a5ec/addons/preview-project-description/userscript.js#L204-L222

**Comment by DNin01** [comment]
So, to summarize, certain events are causing the description preview to re-render, especially after entering (and then leaving) the editor. If this happens in a project that is slowing down your computer, it can also cause the description sections to flicker between the editable and read-only state. These events can supposedly be tracked by watching for `Returning a fake username to mapStateByProps` messages in the console.

**Comment by DNin01** [comment]
Related comment: https://github.com/ScratchAddons/ScratchAddons/pull/6489#issuecomment-1674812259

**Comment by mxmou** [comment]
The problem is that the addon forces a re-render every time the Redux state changes, which happens frequently while a project is running (target and monitor updates). Target updates only cause a state change after entering the editor.

**Comment by WorldLanguages** [comment]
Mxmou's simpler implementation will fix this. Also we could remove the beta tag from the addon.
